mlir/lib/Dialect/Transform/IR/TransformInterfaces.cpp

```diff
@@ -1319,10 +1339,8 @@ void transform::TrackingListener::notifyOperationReplaced(
   // If the op is tracked but no replacement op was found, send a
   // notification.
   if (!diag.succeeded()) {
-    diag.attachNote((*firstAliveUser)->getOwner()->getLoc())
-        << "replacement is required because alive handle(s) exist "
-        << "(first use in this op as operand number "
-        << (*firstAliveUser)->getOperandNumber() << ")";
```

Comment:
I tried to, but it would make the API quite complex. `SkipHandleFn` is pretty simple in the current implementation: is the handle needed or not. It would have to return a user, but the name "skip handle function" would not make sense anymore and we would have to mention the concept of liveness in the `TrackingListener`.